src/main/java/subway/dao/StationDao.java

```diff
@@ -41,14 +41,15 @@ public List<Station> findAll() {
         return jdbcTemplate.query(sql, rowMapper);
     }
 
-    public Station findById(Long id) {
+    public Optional<Station> findById(Long id) {
         String sql = "select * from STATION where id = ?";
-        return jdbcTemplate.queryForObject(sql, rowMapper, id);
+        return jdbcTemplate.query(sql, rowMapper, id)
+            .stream().findAny();
```

Comment:
내부 예외를 활용해 다음처럼 처리하는 방법도 있겠습니다

```suggestion
try {
  return Optional.of(jdbcTemplate.queryForObject(sql, rowMapper));
} catch (EmptyResultDataAccessException e) {
  return Optional.empty();
}

```
<img width="1003" alt="image" src="https://github.com/next-step/spring-subway/assets/51393021/82d39327-76fa-435a-b74c-a099a3038c07">

Reply:
오 이렇게 구현할 수도 있겠군요. 상세한 설명 감사합니다 !